src/basic_types.jl

```diff
@@ -71,14 +71,14 @@ struct BoundaryConditionDoNothing end
                                                 orientation_or_normal_direction,
                                                 direction::Integer, x, t, surface_flux,
                                                 equations)
-    return flux(u_inner, orientation_or_normal_direction, equations)
+    return surface_flux(u_inner, u_inner, orientation_or_normal_direction, equations)
```

Comment:
Why do we need to use `surface_flux` here? isn't that adding computational complexity for the same output?

Reply:
This became necessary, since we now also apply the boundary condition for nonconservative terms. 
When you call `BoundaryConditionDoNothing` with nonconservative terms, the old implementation still evaluates the inner `flux` instead of the inner nonconservative term. Therefore, we erroneously impose the conservative flux twice.
For conservative fluxes this increases the computational complexity, but the modification is necessary to return a consistent boundary condition for both conservative and nonconservative terms.

Reply:
Ok. That makes sense. Thanks!